unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -209,6 +242,16 @@ public boolean equals(final Object other) {
 
   @Override
   public int hashCode() {
-    return Arrays.hashCode(bytes);
+    long result = 1;
```

Comment:
Maybe it's a stupid question, but why do we start with `result = 1` instead of `0`, compared to http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#1493 and http://grepcode.com/file/repository.cloudera.com/content/repositories/releases/org.apache.hadoop.hive/hive-exec/0.7.1-cdh3u1/org/apache/hadoop/hive/ql/udf/generic/GenericUDFHash.java#47

Do we want to cache the hash value like OpenJdk?